corehq/apps/geospatial/utils.py

```diff
@@ -165,3 +167,21 @@ def geojson_to_es_geoshape(geojson):
     es_geoshape = geojson['geometry'].copy()
     es_geoshape['type'] = es_geoshape['type'].lower()
     return es_geoshape
+
+
+def update_cases_owner(domain, case_id_to_owner_id, chunk_size=100):
+    for case_ids in chunked(case_id_to_owner_id.keys(), chunk_size):
```

Comment:
I understand this might need some changes, but it would be better to update related cases together instead of chunking the final list of cases.

Related cases updated together in one form will make more sense for that form and would be useful to look at them together.

Reply:
> Related cases updated together in one form will make more sense for that form and would be useful to look at them together.

Great point. I like the idea of keeping them together.

Just thinking it through, the only reason I think this could be of concern is -  most of the requests in practice seems to have a lower related cases for each case ( < 10). This would create a 100 form submissions with chunk of around 10 cases each which might not be that good performance wise.
Do you think this performance concern is valid?

Reply:
Is it better to update 1000 cases via 1 request or via 10 requests? I guess the right answer would be 1 request if you think of the time/network only. Case updates are safe so updating a high number of cases together is okay.

Though, we should consider other things here as well:
1. the clarity of the update for the case itself. I don't think its unusual for random bulk updates to show up in case history for bulk updates like case imports but there we don't have any relations between cases being updated.
2. number of cases we are restricting to for sync update? Since we are letting user track it, it does not seem harmful if we set a low threshold, if needed, for this to do async updates, when needed, in order to keep things clear/sensible

I like it when data makes sense. And since we are updating related cases, it would be nice to see them together in one form, so that the form makes sense.

So, I agree with your concern. I am thinking if its a concern to address and if yes, how, assuming you find this change request a reasonable effort to make, for something better.

Reply:
Thanks Manish for your suggestions here. Based on it, I did some test to see how much of a concern this i.e. updating related cases really is and here are some findings
- The performance of form submissions is reasonable and since we can have max 100 submissions as a part of this functionality , the performance does not seem to be a strong concern.  As for metrics, I observed on local that 100 forms submissions with one case each took around 4 to 6 seconds.
- In practice, usually a case has up to max 10 related cases. This gives an assurance that this feature would mostly have low data load well under the limit. As for the outliers, the max limit of 5k for this functionality should ensure that this feature does not cause huge load.
- I have reduced the threshold for async to 500 (1k earlier).

The changes have been addressed in d085105be704dfde060e5a6b95ca22f92ffefc85 , 425b0adc6211fdb7e2a58575c88a2e600ba8115a and 6dbea1216ef55bbe755788b16071376cfe5ea035.